### PR TITLE
Astro example

### PR DESCRIPTION
**Pull request by neolivz**
Similar to basic but with two component libraries, one with react and the other with svelte (taken from with-svelte). Both are imported into web, docs Astro projects

### CODE REVIEW

**Comment by neolivz** [comment]
Is there any particular version of pnpm i need to use to generate the lockfile?

**Review by mehulkar — COMMENTED** [review]
Couple q's:

- For @vercel/turbo-oss: are we indefinitely adding examples for frameworks as we get contributions? 
- For @neolivz: Is it expected that the root `pnpm-lock.yaml` changed? I'm not familiar with pnpm, so I'm not sure if it's due to the new example, or if there is an additional update here that is unrelated.

File: .gitignore [hunk 1]
```diff
@@ -2,14 +2,23 @@ node_modules/
...
 *.log
 todos.md
 .store
+
+
+# logs
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+pnpm-debug.log*
```
Comment (nathanhammond) [inline .gitignore:24]: Let's put these changes in a separate PR.

File: .gitignore [hunk 2]
```diff
@@ -2,14 +2,23 @@ node_modules/
...
 .next
 dist/
+.output/
 *.tsbuildinfo
 .eslintcache
 .turbo
 .vercel
 .DS_Store
 .env
+.env.production
```
Comment (nathanhammond) [inline .gitignore:12]:
```suggestion
.env*
```

File: .gitignore [hunk 3]
```diff
@@ -2,14 +2,23 @@ node_modules/
 
 .next
 dist/
+.output/
```
Comment (nathanhammond) [inline .gitignore:5]: I'm not sure what this is for? What is populating this directory?

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -114,7 +114,7 @@ importers:
       eslint-config-next: 12.2.5_ltkjkrq5dazqovqoaeaug2aium
       eslint-config-prettier: 8.5.0_eslint@8.10.0
       postcss: 8.4.16
-      tailwindcss: 3.1.8
+      tailwindcss: 3.1.8_postcss@8.4.16
```
Comment (nathanhammond) [inline pnpm-lock.yaml:117]: Let's remove the lockfile changes from this PR.

File: examples/with-astro/package.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "eslint-config-custom": "workspace:*",
+    "prettier": "latest",
+    "turbo": "latest"
+  },
+  "engines": {
+    "npm": ">=7.0.0",
+    "node": ">=14.0.0"
+  },
+  "dependencies": {},
+  "packageManager": "pnpm@6.25.1"
```
Comment (nathanhammond) [inline examples/with-astro/package.json:25]: We don't specify package manager for any of our examples except for scenarios where it is explicitly required. This is not one of those cases.

File: examples/with-astro/package.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    "lint": "turbo run lint",
+    "format": "prettier --write \"**/*.{ts,tsx,md}\""
+  },
+  "devDependencies": {
+    "eslint-config-custom": "workspace:*",
+    "prettier": "latest",
+    "turbo": "latest"
+  },
+  "engines": {
+    "npm": ">=7.0.0",
```
Comment (nathanhammond) [inline examples/with-astro/package.json:21]:
```suggestion
```

File: examples/with-astro/package.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  ],
+  "scripts": {
+    "build": "turbo run build",
+    "dev": "turbo run dev --parallel",
+    "lint": "turbo run lint",
+    "format": "prettier --write \"**/*.{ts,tsx,md}\""
+  },
+  "devDependencies": {
+    "eslint-config-custom": "workspace:*",
+    "prettier": "latest",
```
Comment (nathanhammond) [inline examples/with-astro/package.json:17]: Let's actually specify a version here.

File: examples/with-astro/package.json [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    "packages/*"
+  ],
+  "scripts": {
+    "build": "turbo run build",
+    "dev": "turbo run dev --parallel",
+    "lint": "turbo run lint",
+    "format": "prettier --write \"**/*.{ts,tsx,md}\""
+  },
+  "devDependencies": {
+    "eslint-config-custom": "workspace:*",
```
Comment (nathanhammond) [inline examples/with-astro/package.json:16]:
```suggestion
    "eslint-config-custom": "*",
```

File: examples/with-astro/packages/tsconfig/package.json [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+  "name": "tsconfig",
+  "version": "0.0.0",
+  "private": true,
+  "main": "index.js",
+  "files": [
+    "base.json",
+    "nextjs.json",
+    "react-library.json"
+  ]
+}
```
Comment (nathanhammond) [inline examples/with-astro/packages/tsconfig/package.json:11]: We just changed our strategy for this, would you mind updating this file?

File: examples/with-astro/README.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
+# Turborepo starter with pnpm
```
Comment (nathanhammond) [inline examples/with-astro/README.md:1]: Many parts of this are not updated, can you correct this? (docs & web are astro, for example)

File: examples/with-astro/apps/docs/astro.config.mjs [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import { defineConfig } from "astro/config";
+import react from "@astrojs/react";
+
+import svelte from "@astrojs/svelte";
```
Comment (nathanhammond) [inline examples/with-astro/apps/docs/astro.config.mjs:4]: Is it expected to use both React and Svelte in an Astro setup?

File: examples/with-astro/apps/docs/README.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+Any static assets, like images, can be placed in the `public/` directory.
+
+## 🧞 Commands
+
+All commands are run from the root of the project, from a terminal:
+
+| Command                | Action                                           |
+| :--------------------- | :----------------------------------------------- |
+| `npm install`          | Installs dependencies                            |
```
Comment (nathanhammond) [inline examples/with-astro/apps/docs/README.md:36]: Since the generator for astro spits out `npm`, we shouldn't use `pnpm`.

**Review by nathanhammond — CHANGES_REQUESTED** [review]
Here's a first pass, hoping that you can sort through these!

**Comment by tknickman** [comment]
Hey @neolivz, apologies for letting this sit for a bit - I'd be happy to take another pass at this and get it in if you have the time to update it!

**Comment by kmalloy24** [comment]
@tknickman I have a working version of this that I can send as a fresh PR if you are interested. 

https://github.com/kmalloy24/with-turbo/tree/main/sveltekit-astro

**Pull request closed without merging**